json/upickle/src/main/scala/sttp/client4/upicklejson/SttpUpickleApi.scala

```diff
@@ -1,29 +1,29 @@
 package sttp.client4.upicklejson
 
-import upickle.default.{read, write, Reader, Writer}
 import sttp.client4._
 import sttp.client4.internal.Utf8
 import sttp.model.MediaType
 import sttp.client4.json._
 
 trait SttpUpickleApi {
+  val upickleApi: upickle.Api
```

Comment:
**question:** Can't we just have a default implementation `= upickle.default` here?

Reply:
we could but that wouldn't be very helpful. The type of the `upickleApi` needs to be overriden to the singleton type of the value (I'll extend the docs). Otherwise, things don't work as expected, as the compiler can't lookup the correct `Reader`/`Writer` values.

Reply:
Oh right, the value type has to be overridden by a concrete type. This dependent type management on the library user side is quite confusing.